assets/src/checkout/pages/BamboraPayPage.tsx

```diff
@@ -65,23 +68,79 @@ const mountFields = (
     .mount(`#${ids.expiry}`);
 };
 
+const normalizeBamboraBrand = (brand: Brand): String => {
+  switch (brand) {
+    case "visa":
+      return CardTypeBrand.Visa;
+    case "mastercard":
+      return CardTypeBrand.MasterCard;
+    case "amex":
+      return CardTypeBrand.AmericanExpress;
+    case "diners":
+      return CardTypeBrand.DinersClub;
+    case "discover":
+      return CardTypeBrand.Discover;
+    case "jcb":
+      return CardTypeBrand.Jcb;
+    default:
+      return "";
+  }
+};
+
+const isCardBrandSupported = (
+  gatewayAccount: GatewayAccountFragment,
+  brand: Brand
+) => {
+  const bamboraBrand = normalizeBamboraBrand(brand);
+  return gatewayAccount.cardTypes.some(
+    cardType => cardType.brand === bamboraBrand
+  );
+};
+
 const bindEventListeners = (
   checkout: CustomCheckout,
-  setCardErrors: React.Dispatch<React.SetStateAction<Record<Field, string>>>
+  setFieldComplete: React.Dispatch<
```

Comment:
Is it possible to come up with a type that represents the current state of our form instead of managing multiple maps of what is going on in each field? it's not clear to me what the difference is between `setFieldError` and `setCardErrors` is. 

maybe something like the following would let us track everything in one place, and then write a couple of helper logic functions to determine whether the card is ready to submit or if an error needs to be shown next to a field or whatever. 

```
type CardStatus {
  brand?: CardTypeBrand | "unsupported";
  [Field]: {error: string} | {ok: true};
}
```

not sure how the [Field] thing would work. need help from J for TS-fu

Reply:
yeah there might be an option here to encapsulate this better. i think we should just revisit this after this PR. are you ok with that @acodegirl or do you want to have a go? i don't mind either way.

Reply:
@jonathaningram Let's revisit this later.I will make a note of it. For now, I will address Peter's first comment on changing the place of usePaymentQuery and update this PR